openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -78,15 +78,11 @@ def __init__(self, course_key, request, username=''):
             username or self.request.user.username,
             course_key,
         )
-        # We must compute course load access *before* setting up masquerading,
-        # else course staff (who are not enrolled) will not be able view
-        # their course from the perspective of a learner.
```

Comment:
I removed this comment, because in my testing, it didn't seem to be true? For the specific callout they made (enrollment), it doesn't actually matter. And for other errors, it also doesn't matter - the lower levels know to check for masquerade status for us.

So I boldly removed this comment, but did not move the lines to avoid any possible regression. And in the course home API, I've done the order differently from this view - I've done masquerading first, then all the lookups.

A) I want to experiment with the masquerading happening first - if successful, it is good data for our future attempts at moving masquerading to some request middleware
B) This only affects staff, so is lowish risk
C) I removed this comment as part of my optimistic push for the order not mattering, and to avoid confusing future folks (if I'm right)

If I'm wrong and this order does matter, we'll maybe get some complaints from staff about masquerading not quite working in some cases, and we can fix and hopefully write more descriptive comments about what exactly fails.